### DISCUSSION TITLE
Ease installation of StreamPipes

### DISCUSSION THREAD

**Discussion by dominikriemer** [apache/streampipes#2046] [Dev]
Currently, we only offer the source release on our website. This requires users to download the full source code, navigate to the installer folder and choose from one of many different installation options.

I think for users who are not developers, it is rather hard to understand that they need to download the full source code. Instead, many project offer other ways to users (e.g., convenience libraries).

Since our installation is based on containers, we cannot simply provide convenience libraries as a direct download. But we could offer a small zip file which only contains the installation files. There could be one docker-compose file at the root and the other installation options in subfolders. 

My idea is to add an additional goal to the streampipes-maven-plugin which runs in the install phase and which creates a small installation zip file containing everything in the installer folder (compose, cli and k8s). There should be one single docker-compose file with a configured broker at the top root of the zip file (maybe a very lightweight protocol such as MQTT or Nats so that the download is as fast as possible). We could provide the zip file on our website and clearly mark it as a convenience library.

What do you think of this idea?

**Comment by bossenti** [comment]
Hi @dominikriemer,

thank you for your suggestion. 
I like the idea of simplifying the installation for first-time users.
A single docker-compose file with a fully configured and lightweight configuration sounds great.

How easy would be such an extension of our Maven plugin?
The provision of this zip file would then also be part of our release process, right?

**Reply by dominikriemer** [reply]
I think that building the zip file is an easy task, but we should put some effort into better explaining the various installation options in a README file and the docs.
The zip file would be saved in the target folder where the source release zip is also located.

**Reply by bossenti** [reply]
Sounds good 👍🏼 

+1 for dedicating some effort in the documentation

**Reply by dominikriemer** [reply]
I just checked our configuration and this can be done even easier with the `maven-assembly-plugin`, so no need for custom code ;-)

**Reply by bossenti** [reply]
This is awesome 🙂

**Comment by bossenti** [comment]
In addition, we could think about offering a one-click deployment as other OSS projects do as well, e.g, [lightdash](https://github.com/lightdash/lightdash?tab=readme-ov-file#1-click-deploy).
How do you think about this?

**Reply by dominikriemer** [reply]
Do you know how this would work with our setup?

**Reply by bossenti** [reply]
In general, you simply need to provide a `render.yaml` in the repository that describes what to start/execute (here is an example https://render.com/docs/blueprint-spec). This would in our case be the docker-compose command.

When someone clicks the button in the README file, render recognizes the repository via the referrer link and executes the deployment in the user's account.

I can give it a try with my private StreamPipes fork first and share my experiences here.

Of course we can also look for other vendors than render
